core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -359,6 +359,26 @@ public List<TopicIdPartition> cachedTopicIdPartitionsInShareSession(String group
         return cachedTopicIdPartitions;
     }
 
+    /**
+     * The acknowledgeShareSessionCacheUpdate method is used to update the share session cache before acknowledgements are handled
+     * either as part of shareFetch request or shareAcknowledge request
+     * @param groupId The group id in the request.
+     * @param memberId The member id of the client in the request.
+     * @param reqEpoch The request epoch.
+     */
+    public void acknowledgeShareSessionCacheUpdate(String groupId, Uuid memberId, int reqEpoch) {
+        // If the request is a Final Fetch Request, the share session would have already been deleted from cache, i.e. there is no need to update the cache
+        if (reqEpoch == ShareFetchMetadata.FINAL_EPOCH) {
+            return;
+        }
+        ShareSession shareSession = cache.get(new ShareSessionKey(groupId, memberId));
```

Comment:
Query: Is it always guranteed that a non-null share session will exist?

Reply:
Removed this method in the latest commit. Pls refer to the reply for above comment for further context. Thanks !